contrib/runners/python_runner/python_runner.py

```diff
@@ -178,14 +179,22 @@ def _get_output_values(self, exit_code, stdout, stderr, timed_out):
         # Parse the serialized action result object
         try:
             action_result = json.loads(action_result)
-        except:
+        except Exception as e:
+            # Failed to de-serialize the result, probably it contains non-simple type or similar
+            LOG.warning('Failed to de-serialize result "%s": %s' % (str(action_result), str(e)))
             pass
 
         if action_result and isinstance(action_result, dict):
             result = action_result.get('result', None)
             status = action_result.get('status', None)
         else:
-            result = 'None'
+            # Failed to de-serialize return result as is aka as a string
+            match = re.search("'result': (.*?)$", action_result).groups()
```

Comment:
Yes, this is horrible but we have a safe-guard below - if there are no results we simply return result as is.

And as mentioned in other comment, I believe that correct long term approach would be to throw.